generator/c2i/c2i_generator.c2

```diff
@@ -16,53 +16,73 @@
 module c2i_generator;
 
 import ast local;
+import component;
 import console;
 import file_utils;
 import string_buffer;
 import string_list;
+import string_pool;
 
 type Generator struct {
+    string_pool.Pool* astPool;
     string_buffer.Buf* out;
+    const char* output_dir;
     string_list.List imports;  // used to filter out duplicate imports (from different ASTs)
     Module* mod;
+    const char* mod_name;
     bool in_body;
 }
 
-public fn void generate(const char* output_dir,
-                        ast.Module* mod)
+public fn void generate(string_pool.Pool* astPool, const char* output_dir, component.List* comps)
 {
-    Generator gen = {
-        .out = string_buffer.create(256*1024, false, 3),
-        .mod = mod,
-        .in_body = false,
+    Generator gen = {}
+    gen.astPool = astPool;
+    gen.out = string_buffer.create(64*1024, false, 3);
+    gen.output_dir = output_dir;
+
+    component.Component* mainComp = comps.getLast();
+
+    // generate external components
+    for (u32 i=0; i<comps.size(); i++) {
+        component.Component* c = comps.get(i);
+        c.visitModules(Generator.on_module, &gen);
```

Comment:
I'm not sure how external modules can be exported, or whether we should allow that. It would be a bit weird, but maybe there is a use-case. If you are creating library1 that depends on some library2 (even in the public symbols), than the .c2i file should contain an import of library2. If depending in a non-public way, then the c2i isn't influenced.